src/video_core/renderer_opengl/renderer_opengl.cpp

```diff
@@ -150,6 +151,14 @@ void RendererOpenGL::SwapBuffers() {
     {
         auto aggregator = Common::Profiling::GetTimingResultsAggregator();
         aggregator->AddFrame(profiler.GetPreviousFrameResults());
+        if (VideoCore::g_toggle_framelimit_enabled) {
+            Common::Profiling::Duration frame_time =
+                aggregator->GetAggregatedResults().frame_time.avg;
+            using LongMs = std::chrono::duration<long, std::chrono::microseconds::period>;
+            std::chrono::microseconds average_frame_time =
+                std::chrono::microseconds(std::chrono::duration_cast<LongMs>(frame_time).count());
+            FrameLimiter(average_frame_time, std::chrono::microseconds(1000000) / 60);
+        }
```

Comment:
@JayFoxRox the previous implementation was faster(maintained a frame_time as close to 16.666ms as possible).i tried reducing the number of casts, to this
`if (VideoCore::g_toggle_framelimit_enabled) {
            Common::Profiling::Duration frame_time =
                aggregator->GetAggregatedResults().frame_time.avg;
            //using LongMs = std::chrono::duration<long, std::chrono::microseconds::period>;
            std::chrono::microseconds average_frame_time = std::chrono::duration_cast<std::chrono::microseconds>(frame_time);
            FrameLimiter(average_frame_time, std::chrono::microseconds(1000000)/60);`
no improvement there.
i could pass both parameters as float, and let the casting be in the function definition, so as not to execute extra instruction before the decision.